File: 01 Face Recognition/Python/toCreateDatabase/OpenCv/conditionImage.py
#import cv2
#import cv2.cv as cv
import numpy as np
import random
import math
import os
import PIL                  #CAREFUL check if available for the board
from PIL import Image
import ImageOps

classes = 2
trainImg = 6

numRows = 92
numCols = 112

fileRead = 'D:\\Final Year Project\\Code\\RealWorldDatabase\\Original\\s'
pathWrite = 'D:\\Final Year Project\\Code\\RealWorldDatabase\\Conditioned\\s'


for classCounter in range(1,classes+1):
    path = pathWrite + str(classCounter)
    os.mkdir(path)
    for trainImgCounter in range(1,trainImg+1):
        rd = fileRead + str(classCounter) + '\\' + str(trainImgCounter) + '.jpg'
        wr = pathWrite + str(classCounter) + '\\' + str(trainImgCounter) + '.jpg'
        dst = Image.open(rd)
        dst = dst.resize((numRows,numCols))
        dst = ImageOps.grayscale(dst)
        #dst.show()
        dst.save(wr)

print 'CONDITIONED'

File: 01 Face Recognition/Matlab Codes/conditionImage.m
clear all
clc
close all

classes = 1;
trainImg = 7;

numRows = 112;
numCols = 92;

dirPath = 'D:\Final Year Project\Code\RealWorldDatabase\Conditioned\s';
imgReadPath = 'D:\Final Year Project\Code\RealWorldDatabase\Original\s';
imgWritePath = 'D:\Final Year Project\Code\RealWorldDatabase\Conditioned\s';

i=1;
j=7;
% folderName = 'Conditioned';
% mkdir(strcat('D:\Final Year Project\Code\RealWorldDatabase\',folderName));
%for i = 1 : classes
 %   mkdir(strcat(dirPath,int2str(classes)));
  %  for j = 1 : trainImg
  
       img = imread(strcat(imgReadPath,int2str(i),'\',int2str(j),'.jpg')); 
       img = rgb2gray(img);
       img = imresize(img,[numRows numCols]);
       imwrite(img,strcat(imgWritePath,int2str(i),'\',int2str(j),'.jpg'));
    %end
%end

display('Done')
       
        